dbt-tests-adapter/dbt/tests/adapter/incremental/test_incremental_microbatch.py

```diff
@@ -0,0 +1,84 @@
+import os
+from unittest import mock
+
+import pytest
+from freezegun import freeze_time
+
+from dbt.tests.util import relation_from_name, run_dbt
+
+_input_model_sql = """
+{{ config(materialized='table', event_time='event_time') }}
+select 1 as id, TIMESTAMP '2020-01-01 00:00:00-0' as event_time
+union all
+select 2 as id, TIMESTAMP '2020-01-02 00:00:00-0' as event_time
+union all
+select 3 as id, TIMESTAMP '2020-01-03 00:00:00-0' as event_time
+"""
+
+_microbatch_model_sql = """
+{{ config(materialized='incremental', incremental_strategy='microbatch', unique_key='id', event_time='event_time', batch_size='day') }}
+select * from {{ ref('input_model') }}
+"""
+
+
+class BaseMicrobatch:
+    @pytest.fixture(scope="class")
+    def microbatch_model_sql(self):
+        return _microbatch_model_sql
+
+    @pytest.fixture(scope="class")
+    def input_model_sql(self):
+        return _input_model_sql
+
+    @pytest.fixture(scope="class")
+    def models(self, microbatch_model_sql, input_model_sql):
+        return {
+            "input_model.sql": input_model_sql,
+            "microbatch_model.sql": microbatch_model_sql,
+        }
+
+    def assert_row_count(self, project, relation_name: str, expected_row_count: int):
```

Comment:
I struggle with having assertions outside of the test method. I wind up having to track them down when they fail. Could we instead convert this to a method that obtains the row count? We'd wind up with something like this:

```python
...

def row_count(self, project, relation_name: str) -> int:
    relation = relation_from_name(project.adapter, relation_name)
    result = project.run_sql(f"select count(*) as num_rows from {relation}", fetch="one")
    return result[0]

...

def test_run_with_event_time(self, project):
    ...
    with freeze_time("2020-01-03 14:57:00"):
        run_dbt(["run", "--select", "microbatch_model"])
    assert self.row_count(project, "microbatch_model") == 3
    ...
```

Reply:
did something slightly different so we have consistent assertion failure message as part of the helper